lib/Echidna/ABI.hs

```diff
@@ -221,9 +226,9 @@ mutateValue (AbiArray s t a) =
 changeOrId :: (Traversable t, MonadGen m) => (a -> m a) -> t a -> m (t a)
 changeOrId f = mapM $ (Gen.element [f, pure] >>=) . (&)
 
-mutateCall :: MonadGen m => SolCall -> m SolCall
-mutateCall (t, vs) = (t,) <$> changeOrId mutateValue vs
+mutateCall :: MonadGen m => Config -> SolCall -> m SolCall
```

Comment:
let's move this back to a `MonadReader` and use `data-has` to keep this general, then everything in this file has the same style